src/timer.c

```diff
@@ -15,141 +13,80 @@
 #include "timer.h"
 
 #if __MACH__
-static uint64_t sm_timebase;
+mach_timebase_info_data_t sm_timebase;
```

Comment:
Thinking out loud:
I see you now use the `sm_timebase` variable from the macros, which is why you dropped `static`, but do we need to use those macros from outside of `timer.[ch]`? OK, I guess it's safer that way, so that we don't get a build failure on macOS later if someone starts using those macros externally yet only tests their changes on other systems.

What does the `sm` prefix stand for? Maybe it's better to use some prefix common for everything we have in `timer.h`.